tests/zfs-tests/tests/functional/cli_root/zdb/zdb_display_block.ksh

```diff
@@ -0,0 +1,128 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019 by Datto, Inc. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# Description:
+# zdb -R pool <DVA>:b will display the block
+#
+# Strategy:
+# 1. Create a pool, set compression to lzjb
+# 2. Write some identifiable data to a file
+# 3. Run zdb -ddddddbbbbbb against the file
+# 4. Record the DVA of the first L1 block;
+#    record the first L0 block display; and
+#    record the 2nd L0 block display.
+# 5. Run zdb -R with :bd displays first L0
+# 6. Run zdb -R with :b80d displays 2nd L0
+# 7. Run zdb -R with :db80 displays 2nd L0
+# 8. Run zdb -R with :id flag displays indirect block
+#     (similar to zdb -ddddddbbbbbb output)
+# 9. Run zdb -R with :id flag and .0 vdev
+#
+
+
+function cleanup
+{
+	datasetexists $TESTPOOL && destroy_pool $TESTPOOL
+}
+
+log_assert "Verify zdb -R :b flag (block display) works as expected"
+log_onexit cleanup
+init_data=$TESTDIR/file1
+write_count=256
+blksize=4096
+
+# only read 256 128 byte block pointers in L1 (:i flag)
+# 256 x 128 = 32k / 0x8000
+l1_read_size="8000"
+
+verify_runnable "global"
+verify_disk_count "$DISKS" 2
+
+default_mirror_setup_noexit $DISKS
+log_must zfs set recordsize=$blksize $TESTPOOL/$TESTFS
+log_must zfs set compression=lzjb $TESTPOOL/$TESTFS
+
+file_write -d R -o create -w -f $init_data -b $blksize -c $write_count
+sync_pool $TESTPOOL true
+
+# get object number of file
+listing=$(ls -i $init_data)
+set -A array $listing
+obj=${array[0]}
+log_note "file $init_data has object number $obj"
+
+output=$(zdb -ddddddbbbbbb $TESTPOOL/$TESTFS $obj 2> /dev/null \
+    |grep -m 1 "L1  DVA" |head -n1)
+dva=$(grep -oP 'DVA\[0\]=<\K.*?(?=>)' <<< "$output")
```

Comment:
For compatibility with FreeBSD please avoid using the Perl extensions with grep. This looks the same as what was changed in 6f481612f37a4cffaff142b80f6a4c37257406c6 for an example of an alternative that works on FreeBSD.

Reply:
Because of multiple DVA copies I think the suggested sed method needs another filter on it (sed is taking final '>' as end delimiter).

```
$ export output="0 L1 DVA[0]=<0:40109600:2000> DVA[1]=<0:50000800:2000> [L1 ZFS plain file]"
$ echo $output | sed -Ene 's/^.+DVA\[0\]=<(.+)>.*/\1/p'
0:40109600:2000> DVA[1]=<0:50000800:2000
$ echo $output | sed -Ene 's/^.+DVA\[0\]=<(.+)>.*/\1/p' | awk '{split($0,array,">")} END{print array[1]}'
0:40109600:2000
```

Reply:
How about this and I will update the existing instances to the same:
```
$ echo $output | sed -Ene 's/^.+DVA\[0\]=<([^>]+)>.*/\1/p'
0:40109600:2000
```

Reply:
Nice.  You speak fluent sed!  Will update zdb_display_block.ksh here.